src/esp/sim/Simulator.cpp

```diff
@@ -117,171 +126,376 @@ void Simulator::reconfigure(const SimulatorConfiguration& cfg) {
-      setNavMeshVisualization(false);  // first clear the old instance
-      setNavMeshVisualization(true);
+  // whether or not to correct for COM shift - only do for blender-sourced scene
+  // attributes
+  bool Default_COM_Correction =
+      (static_cast<metadata::managers::SceneInstanceTranslationOrigin>(
+           curSceneInstanceAttributes->getTranslationOrigin()) ==
+       metadata::managers::SceneInstanceTranslationOrigin::AssetLocal);
+
+  // Iterate through instances, create object and implement initial
+  // transformation.
+  for (const auto& objInst : objectInstances) {
+    const std::string objAttrFullHandle =
+        metadataMediator_->getObjAttrFullHandle(objInst->getHandle());
+    if (objAttrFullHandle == "") {
+      LOG(WARNING) << "Simulator::createSceneInstance : Unable to find object "
+                      "attributes whose handle contains "
+                   << objInst->getHandle()
+                   << " as specified in object instance attributes, so unable "
+                      "to instance object; skipping. ";
+      continue;
     }
+    objID = physicsManager_->addObject(objAttrFullHandle, &drawables,
+                                       attachmentNode, lightSetupKey);
+    if (objID == ID_UNDEFINED) {
+      // instancing failed for some reason.
+      LOG(WARNING)
+          << "Simulator::createSceneInstance : Failed to instantiate object "
+             "specified in Scene Instance Attributes using template named : "
+          << objInst->getHandle();
+      continue;
+    }
+    // set object's location and rotation based on translation and rotation
+    // params specified in instance attributes
+    auto translate = objInst->getTranslation();
+    // get instance override value, if exists
+    auto Instance_COM_Origin =
+        static_cast<metadata::managers::SceneInstanceTranslationOrigin>(
+            objInst->getTranslationOrigin());
+    if (((Default_COM_Correction) &&
+         (Instance_COM_Origin !=
+          metadata::managers::SceneInstanceTranslationOrigin::COM)) ||
+        (Instance_COM_Origin ==
+         metadata::managers::SceneInstanceTranslationOrigin::AssetLocal)) {
+      // if default COM correction is set and no object-based override, or if
+      // Object set to correct for COM.
+
+      translate -= objInst->getRotation().transformVector(
+          physicsManager_->getObjectVisualSceneNodes(objID)[0]->translation());
+    }
+    physicsManager_->setTranslation(objID, translate);
+    physicsManager_->setRotation(objID, objInst->getRotation());
+    // set object's motion type if different than set value
+    const physics::MotionType attrObjMotionType =
+        static_cast<physics::MotionType>(objInst->getMotionType());
+    const physics::MotionType objMotionType =
+        physicsManager_->getObjectMotionType(objID);
+
+    if ((attrObjMotionType != objMotionType) &&
+        (attrObjMotionType != physics::MotionType::UNDEFINED)) {
```

Comment:
The setter will check for equivalency before doing anything, so you could simplify this.

Reply:
I was thinking of doing that, but I wanted it visually obvious to implementers/supporters the exact conditions where this would be executed.